### ISSUE TITLE
Regarding sandbox.js and test setup

### ISSUE THREAD

**Issue by omerjava** [lab-antwerp-1/home#308]
In [Logic (our project folder)](https://github.com/lab-antwerp-1/group1-separate-of-concerns-project/tree/master/client/src/logic) folder of separation-of-concerns-starter template, there is `sandbox.js` file. I am curious about the exact purpose of this file. In our project so far, there are 2 logic JS files. I wrote a test codes for each of them and saved them as `.test.js` files. Then, I had to setup 'Jest' inside the 'logic' folder with [npm install --save-dev jest](https://jestjs.io/docs/getting-started) and run test locally in this folder. In this phase, there happened import/export problem between `isBookNotInTheList.js` and `isBookNotInTheList.test.js` because 'Jest' test run gave error with normal JS import method. I had to use node `module.exports` method instead. But, this time `isBookNotInTheList.js` was problem, because exported the logic function in this file must be used by handlers. As a solution, I copied logic functions to `sandbox.js` and used `module.exports` method to export them to the test files. Test files worked well with `npm run test` as a result. So, my first question, is this the duty of `sandbox.js` file? To collect all logic functions in one place, isolate them and export them to relevant test files. If not, what is the role/duty of it?

My second question is about running test. When I first wrote a test as a `.spec.test` file and tried `npm run test -- ./client`, I received this error `basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")  ^^^^^^^ SyntaxError: missing ) after argument list` .
I made some research in Google and couldn't find an applicable solution although there was a lot of pages about this error. As a result, I set up 'Jest' in logic folder with `node_modules` and `package.json`, then it worked. So, is there a solution to above-mentioned error? And is it good idea to setup 'Jest' in a specific folder in a project? Maybe I need to setup it again to components folder. It looks a bad practice.

**Comment by colevandersWands** [comment]
The purpose of `sandbox.js` is just somewhere for you to do experiments while you're developing.  no more no less.

And for testing with Jest, you don't need to install that separately in the logic folder.  It's already configured for the repository to work with `import`.  There's instructions for using that script in the README

**Comment by colevandersWands** [comment]
The purpose of `sandbox.js` is to have somewhere you can do experiments while writing your logic, no more no less.

**Comment by colevandersWands** [comment]
To make Jest work with `import` you need to set the `--experimental-vm-modules` flag, like in [the npm script](https://github.com/lab-antwerp-1/group1-separate-of-concerns-project/blob/5a3c17e6e3e5d0ee5bc523a3d50cfea84be764c4/package.json#L8).

**Comment by colevandersWands** [comment]
I just cloned, installed and ran the tests locally with `npm run test -- ./client`.  They all passed. maybe it's your machine?

**Comment by omerjava** [comment]
> I just cloned, installed and ran the tests locally with `npm run test -- ./client`. They all passed. maybe it's your machine?
> 
> I'm on a mac, are you on windows? if you are on windows maybe try `npm run test -- client` without the `/` or with `\client`

Yes, I use Windows. I will try your suggestions. Thank you very much for replies

**Comment by omerjava** [comment]
It gave same error with all these `npm run test -- client` and `npm run test -- \client`  and `npm run test -- .\client`

**Comment by omerjava** [comment]
I changed `"test": "node --experimental-vm-modules node_modules/.bin/jest --passWithNoTests"` as `"test": "jest"` and I didn't change this line `"ci:test": "node --experimental-vm-modules node_modules/.bin/jest --passWithNoTests ./client"`. Now, it is working when `npm run test -- ./client`
